src/core/io/src/4C_io_vtu_reader.cpp

```diff
@@ -0,0 +1,215 @@
+// This file is part of 4C multiphysics licensed under the
+// GNU Lesser General Public License v3.0 or later.
+//
+// See the LICENSE.md file in the top-level for license information.
+//
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#include "4C_io_vtu_reader.hpp"
+
+#include "4C_fem_general_cell_type.hpp"
+#include "4C_fem_general_cell_type_traits.hpp"
+#include "4C_io_element_vtk_cell_type_register.hpp"
+#include "4C_utils_exceptions.hpp"
+
+#ifdef FOUR_C_WITH_VTK
+#include <vtkArrayDispatch.h>
+#include <vtkCell.h>
+#include <vtkCellData.h>
+#include <vtkDataArray.h>
+#include <vtkIdList.h>
+#include <vtkLongLongArray.h>
+#include <vtkPointData.h>
+#include <vtkSmartPointer.h>
+#include <vtkType.h>
+#include <vtkUnstructuredGrid.h>
+#include <vtkXMLUnstructuredGridReader.h>
+#include <vtkXMLUnstructuredGridWriter.h>
+#endif
+#include <algorithm>
+
+
+FOUR_C_NAMESPACE_OPEN
+
+#ifdef FOUR_C_WITH_VTK
+namespace
+{
+  // Returns a reference to the specified array in the given vtkDataArrayCollection
+  vtkDataArray& get_array(auto* data, const std::string& name)
+  {
+    vtkDataArray* data_array = data->GetArray(name.c_str());
+    FOUR_C_ASSERT_ALWAYS(data_array, "Array {} not found!", name);
```

Comment:
The error could be improved to tell users that 4C requires special arrays (as you say in the docs) and what they should contain.